### PR TITLE
feat(ui): Deprioritize "display name" settings fields

### PR DESCRIPTION
**Pull request by billyvg**
Any suggestions for copy here?

image

### CODE REVIEW

**Comment by dcramer** [comment]
*will be used

File: src/sentry/static/sentry/app/data/forms/teamSettingsFields.jsx [hunk 1]
```diff
@@ -7,22 +11,26 @@ const formGroups = [
...
+        name: 'name',
         type: 'string',
         required: true,
-        label: 'Short Name',
-        placeholder: 'e.g. api-team',
-        help: 'A unique ID used to identify the team',
+        label: t('Legacy Name'),
+        placeholder: 'e.g. API Team',
+        help: tct(
+          '[Deprecated] In the future, only [Name] will be used to identify your project',
```
Comment (macqueen) [inline src/sentry/static/sentry/app/data/forms/teamSettingsFields.jsx]: oh and project should be team

File: src/sentry/static/sentry/app/data/forms/organizationGeneralSettings.jsx [hunk 2]
```diff
@@ -11,20 +12,25 @@ const formGroups = [
...
-        name: 'slug',
+        name: 'name',
         type: 'string',
         required: true,
-        label: t('Short Name'),
-        help: t('A unique ID used to identify this organization.'),
+
+        label: t('Legacy Name'),
+        help: tct(
+          '[Deprecated] In the future, only [Name] will be used to identify your project',
```
Comment (macqueen) [inline src/sentry/static/sentry/app/data/forms/organizationGeneralSettings.jsx]: lol and project should be org

**Review by macqueen — APPROVED** [review]

**Comment by dcramer** [comment]
#5741

**Pull request merged**